test/e2e/tests/ppom/ppom-blockaid-alert-contract-interaction.spec.js

```diff
@@ -0,0 +1,244 @@
+      ],
+    })
+    .thenCallback(async (req) => {
+      return {
+        statusCode: 200,
+        json: {
+          jsonrpc: '2.0',
+          id: (await req.body.getJson()).id,
+          error: {
+            code: -32601,
+            message:
+              'The method debug_traceCall does not exist/is not available',
+          },
+        },
+      };
+    });
+
+  await mockServer
+    .forGet('https://www.4byte.directory/api/v1/signatures/')
+    .always()
+    .withQuery({ hex_signature: '0xef5cfb8c' })
+    .thenCallback(() => ({
+      statusCode: 200,
+      json: {
+        count: 1,
+        next: null,
+        previous: null,
+        results: [
+          {
+            id: 187294,
+            created_at: '2021-05-12T10:20:16.502438Z',
+            text_signature: 'claimRewards(address)',
+            hex_signature: '0xef5cfb8c',
+            bytes_signature: 'ï\\û',
+          },
+        ],
+      },
+    }));
+}
+
+describe('PPOM Blockaid Alert - Malicious Contract interaction @no-mmi', function () {
+  it('should show banner alert', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withNetworkControllerOnMainnet()
+          .withPermissionControllerConnectedToTestDapp({
+            useLocalhostHostname: true,
+          })
+          .withPreferencesController({
+            securityAlertsEnabled: true,
+            preferences: {
+              redesignedTransactionsEnabled: true,
+              redesignedConfirmationsEnabled: true,
+              isRedesignedConfirmationsDeveloperEnabled: true,
+            },
+          })
+          .build(),
+        defaultGanacheOptions,
+        testSpecificMock: mockInfura,
+        title: this.test.fullTitle(),
+      },
+
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        await openDapp(driver, null, 'http://localhost:8080');
+
+        const expectedTitle = 'This is a deceptive request';
+        const expectedDescription =
+          'If you approve this request, a third party known for scams will take all your assets.';
+
+        // Click TestDapp button to send JSON-RPC request
+        await driver.clickElement('#maliciousContractInteractionButton');
+
+        // Wait for confirmation pop-up
+        await driver.waitUntilXWindowHandles(3);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        await driver.assertElementNotPresent('.loading-indicator');
+
+        const bannerAlert = await driver.findElement(
+          '[data-testid="confirm-banner-alert"]',
+        );
+
+        const bannerAlertText = await bannerAlert.getText();
+
+        assert.ok(
+          bannerAlertText.includes(expectedTitle),
+          `Expected text "${expectedTitle}" not found on the alert`,
+        );
+        assert.ok(
+          bannerAlertText.includes(expectedDescription),
+          `Expected text "${expectedDescription}" not found on the alert`,
```

Comment:
hi there :wave: this is an anti-pattern that we are trying to remove from our e2e. Could we change it, so we find the element by the expected text instead? :pray:  See more info around this [here](https://github.com/MetaMask/metamask-extension/issues/19870)

Reply:
thanks for pointing it out, it's now changed!